### ISSUE TITLE
Problem in restoration.denoise_bilateral with zeros image

### ISSUE THREAD

**Issue by odebeir** [scikit-image/scikit-image#1677]
it seems that the special case of a null image returns

> ValueError: The maximum value found in the image was 0.

``` python

from skimage.restoration import denoise_bilateral
import numpy as np

img0 = np.zeros((10,10))
img1 = np.ones((10,10))

#this pass
print denoise_bilateral(img1, sigma_range=0.05, sigma_spatial=15)

# this breaks
print denoise_bilateral(img0, sigma_range=0.05, sigma_spatial=15)
```

**Comment by emmanuelle** [comment]
@odebeir thank you for spotting this bug. There is a comment saying in the source code of the cython function
```
   # if image.max() is 0, then dist_scale can have an unverified value
   # and color_lut[<int>(dist * dist_scale)] may cause a segmentation fault
   # so we verify we have a positive image and that the max is not 0.0.
```
but in fact we only check for strictly negative values
```
if image.min() < 0.0:
        raise ValueError("Image must contain only positive values")
```
So this should we replaced by 
```
if image.min() <= 0.0:
        raise ValueError("Image must contain only positive values")
```

**Comment by emmanuelle** [comment]
Also, I can't see any good reason why `denoise_bilateral` should not work on images with negative values. How about substracting the minimum of the image?

**Comment by odebeir** [comment]
indeed, or maybe a mention in the doc of the function could mention the range of accepted values

**Comment by kshitij10496** [comment]
Is this issue still open ?

**Comment by soupault** [comment]
@kshitij10496 It is. Would you like to investigate it?

**Comment by kshitij10496** [comment]
@soupault Yeah.
I would love to work on this

**Comment by kshitij10496** [comment]
I really appreciate the work of @odebeir here.
However, I have a few queries regarding this:

- I was looking into the `_denoise_cy.pyx` file and I found [this](https://github.com/scikit-image/scikit-image/blob/master/skimage/restoration/_denoise_cy.pyx#L59-L61).  
Does this mean that adding support for negative images may lead to further bugs in the codebase ?
- Should we just restrict to the case when the image is uniform ( like @ahojnnes suggested : `image.max() == image.min()`) irrespective of the values ?

Since, I am new here, I would appreciate advices.

Ping @soupault @emmanuelle @ahojnnes !

**Comment by soupault** [comment]
Closed via #2533 .
